File: Server/route/challengeClass.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetChallengeClassRoute(r *gin.Engine) {
	challengeClass := r.Group("/challengeClass")
	challengeClass.Use(middleware.JWTAuthMiddleware())
	{
		challengeClass.POST("/GetChallengeClassList", api.GetChallengeClassList)
		challengeClass.POST("/GetChallengeClass", api.GetChallengeClass)
	}
	adminChallengeClass := r.Group("/challengeClass")
	adminChallengeClass.Use(middleware.AdminAuthMiddleware())
	{
		adminChallengeClass.POST("/CreateChallengeClass", api.CreateChallengeClass)
		adminChallengeClass.POST("/UpdateChallengeClass", api.UpdateChallengeClass)
		adminChallengeClass.POST("/DeleteChallengeClass", api.DeleteChallengeClass)
	}
}

File: Web/src/api/challengeClass.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetChallengeClassList() {
  return apiClient.post("/challengeClass/GetChallengeClassList").then(handleResponse).catch(handleError)
}

function GetChallengeClass(data) {
  return apiClient.post("/challengeClass/GetChallengeClass", data).then(handleResponse).catch(handleError)
}

function CreateChallengeClass(data) {
  return apiClient.post("/challengeClass/CreateChallengeClass", data).then(handleResponse).catch(handleError)
}

function UpdateChallengeClass(data) {
  return apiClient.post("/challengeClass/UpdateChallengeClass", data).then(handleResponse).catch(handleError)
}

function DeleteChallengeClass(data) {
  return apiClient.post("/challengeClass/DeleteChallengeClass", data).then(handleResponse).catch(handleError)
}

const challengeClassApi = {
  GetChallengeClassList,
  GetChallengeClass,
  CreateChallengeClass,
  UpdateChallengeClass,
  DeleteChallengeClass
}

export default challengeClassApi
